### PR TITLE
use serialize object

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -211,16 +189,16 @@ class Logger {
       message = tags;
       tags = [];
     }
-    message = this.serialize(tags, message, this.config);
     if (this.config.defaultTags.length !== 0) {
       tags = this.config.defaultTags.concat(tags);
     }
+    const serializedMsg = typeof message === 'object' ? this.serialize(tags, message, this.config) : message;
```
Comment (jgallen23) [inline index.js]: why did this change?

File: index.js [hunk 2]
```diff
@@ -211,16 +189,16 @@ class Logger {
       message = tags;
       tags = [];
     }
-    message = this.serialize(tags, message, this.config);
     if (this.config.defaultTags.length !== 0) {
       tags = this.config.defaultTags.concat(tags);
     }
+    message = typeof message === 'object' ? this.serialize(tags, message, this.config) : message;
```
Comment (jgallen23) [inline index.js]: what does this do?

Comment (orthagonal) [reply index.js]: serialize-object expects an object and will error if you pass a string, I feel like we should push this check into serialize-object.

Comment (jgallen23) [reply index.js]: yeah, ideally we shouldn't be changing any other logic other than swapping the function with the library

Comment (jgallen23) [reply index.js]: lets try to make it so we don't really have to change anything else and just swap the function with the lib

File: index.js [hunk 3]
```diff
@@ -215,8 +190,9 @@ class Logger {
     if (this.config.defaultTags.length !== 0) {
       tags = this.config.defaultTags.concat(tags);
     }
+    // message = typeof message === 'object' ? this.serialize(tags, message, this.config) : message;
     Object.keys(this.reporters).forEach((name) => {
-      const messageClone = (typeof message === 'object') ? aug(message) : message;
+      const messageClone = (typeof message === 'object') ? Object.assign({}, message) : message;
```
Comment (jgallen23) [inline index.js]: why did you change this?

Comment (jgallen23) [reply index.js]: Object.assign doesn't do a deep clone
